docs/source/deploy_chaincode.md

```diff
@@ -0,0 +1,348 @@
+
+We could have provided a ``--signature-policy`` or ``--channel-config-policy`` argument to the chaincode approval to specify a chaincode endorsement policy. The endorsement policy specifies how many peers belonging to different channel members need to validate a transaction against a given chaincode. Because we did not set a policy, the definition of Fabcar will use the default endorsement policy, which requires that a transaction be endorsed by a majority of channel members present when the transaction is submitted. This implies that if new organizations are added to or removed from the channel, the endorsement policy
+is updated automatically to require more or fewer endorsements. In this tutorial, the default policy will require a majority of 2 out of 2 and transactions will need to be endorsed by a peer from Org1 and Org2. If you want to specify a custom endorsement policy, you can use the [Endorsement Policies](endorsement-policies.html) operations guide to learn about the policy syntax.
+
+The `peer lifecycle chaincode approveformyorg` command needs to be submitted and signed as your organization admin identity. As a result, the `CORE_PEER_MSPCONFIGPATH` variable needs to point to the MSP folder of an admin identity. You cannot approve a chaincode definition with a client user. The approval needs to be submitted to the ordering service, which will validate the admin signature and then distribute the approval to your peers.
+
+We still need to approve the chaincode definition as Org1. Set the following environment variables to operate as the Org1 admin:
+```
+CORE_PEER_LOCALMSPID="Org1MSP"
+CORE_PEER_MSPCONFIGPATH=${PWD}/organizations/peerOrganizations/org1.example.com/users/Admin@org1.example.com/msp
+CORE_PEER_TLS_ROOTCERT_FILE=${PWD}/organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls/ca.crt
+CORE_PEER_ADDRESS=localhost:7051
+```
+
+You can now approve the chaincode definition as Org1.
+```
+peer lifecycle chaincode approveformyorg -o localhost:7050 --ordererTLSHostnameOverride orderer.example.com --channelID mychannel --name fabcar --version 1.0 --init-required --package-id $CC_PACKAGE_ID --sequence 1 --tls true --cafile ${PWD}/organizations/ordererOrganizations/example.com/orderers/orderer.example.com/msp/tlscacerts/tlsca.example.com-cert.pem
+```
+
+We now have the majority we need to deploy the Fabcar the chaincode to the channel. While only a majority of organizations need to approve a chaincode definition (with the default policies) all organizations need to approve a chaincode definition to start the chaincode on their peers. As a result, it is recommended that all channel members approve a chaincode before committing the chaincode definition.
```

Comment:
This is a really important point that is easy to miss.

While we have satisfied the cc lifecycle endorsement policy which means that the cc is now permitted to be committed to the channel, if there are other peers in the organizations or additional organizations with peers on the channel, do they all need to approve the chaincode?

Does every peer in every org need to run the approve, or at least 1 peer in every org on the channel?

Reply:
Added another sentance for emphasis: If you commit the definition before a channel member has approved the chaincode, the organization will not be able to endorse transactions.

I say that approval is at the org level above and below.